src/main/java/jetbrains/buildServer/iaa/utils/CustomParameters.java

```diff
@@ -43,4 +48,12 @@ private static int parseThreshold(@Nullable String value) {
     int parsedValue = StringUtil.parseInt(value, DEFAULT_TEST_COUNT_THRESHOLD);
     return parsedValue >= 0 ? parsedValue : Integer.MAX_VALUE;
   }
+
+  public boolean isSilentModeOn(@NotNull SBuild build) {
+    Collection<SBuildFeatureDescriptor> descriptors = build.getBuildFeaturesOfType(Constants.BUILD_FEATURE_TYPE);
+    if (descriptors.isEmpty()) throw new IllegalStateException("Descriptors should not be empty");
```

Comment:
Is it possible that this code will be executed on build without auto-assigned feature?

Reply:
No, from my point of view it is not possible. When we process the buildProblemsChanged event we check this.